src/lib/box.js

```diff
@@ -1475,7 +1475,7 @@ foam.CLASS({
         req.then(function(resp) {
           return resp.payload;
         }).then(function(p) {
-          this.me.send(foam.json.parse(foam.json.parseString(p, null, this)));
+          this.me.send(foam.json.parse(foam.json.parseString(p, this, this)));
```

Comment:
This line should actually just read

this.me.send(foam.json.parseString(p, this));

since parseString() calls parse() internally.  When I wrote this originally parseString() just did the eval() portion and didn't call parse(), or at least that's what I thought when I first wrote it.

Doing this should also remove the need for your nested try{}catch{} earlier.